Readme update to reflect v3 scripts too (#208)
broadinstitute/gnomad_qc: gnomad_qc/v3/README.md

## Contents and ordering of scripts
The scripts below are run approximately in the order they are listed. We begin quality control with a sparse MT created using Hail's [gVCF combiner](https://hail.is/docs/0.2/experimental/vcf_combiner.html)

### Quality Control
**load_data/**
* `create_last_END_positions.py` — Compute the genomic position of the most upstream reference block overlapping each row on the raw sparse MatrixTable.

**sample_qc/**
* `sample_qc.py`
  * `--sample_qc` - Compute Hail's sample QC metrics on the raw split MatrixTable stratified by bi-allelic and multi-allelic variants.
  * `--impute_sex` - Impute chromosomal sex karyotype annotation.
  * `--compute_hard_filters` - Determine the samples that fail hard filtering thresholds.

>   * `--compute_samples_ranking` - Compute global sample ranking based on hard-filters, releasable and coverage. This is used by `--compute_related_samples_to_drop` to determine sample removal due to relatedness.

Author: oops, I will remove from both places, I combined this into compute_related_samples_to_drop and apparently didn't remove the argument

* `generate_qc_annotations.py`
  * `--generate_allele_data` - Determine the following annotations for each variant in the split info Table: variant type (SNV, indel, multi-SNV, multi-indel, mixed), allele type (SNV, insertion, deletion, complex), and the total number of alleles present at the site.
  * `--generate_ac` - Allele count per variant (raw and adj filtered genotypes) of high quality samples, high quality unrelated samples, and release samples.
  * `--generate_fam_stats` - Calculate transmission and de novo statistics using trios (`sample_qc.py --finalize_ped`).
  * `--vep` - Get the Ensembl Variant Effect Predictor (VEP) annotation for each variant.
  * `--export_transmitted_singletons_vcf` - Export transmitted singletons (`--generate_fam_stats`) to VCF files, used for running VQSR with transmitted singletons in the training.

**create_release/**
* `create_release_sites_ht.py` - Combine frequency, filtering allele frequency, variant QC, VEP, dbSNP, in silico scores, and variant QC metric histogram annotations into unified table. Add frequency annotations for each sample subset (e.g., controls, non-neuro, non-cancer, non-TOPMED, non-v2, tgp, hgdp).

> * `create_hgdp_tgp_subset.py` - Subset raw sparse MatrixTable to only samples in the Human Genome Diversity Project (HGDP) and 1000 genomes project (1KG/TGP) to create an unsplit sparse MatrixTable and split dense MatrixTable (with full sample and variant annotations) that can be made publicly available.

Reviewer: ```suggestion
* `create_hgdp_tgp_subset.py` - Subset raw sparse MatrixTable to only samples in the Human Genome Diversity Project (HGDP) and 1000 Genomes Project (1KG/TGP) to create an unsplit sparse MatrixTable and split dense MatrixTable (with full sample and variant annotations) that can be made publicly available.
```

> * `prepare_vcf_data_release.py` - Select and reformat release annotations for VCF export, create VCF header text, and perform sanity check on release annotations.

Author: Yup, it's coming soon hopefully though so I figured I would just add it

* `make_var_annot_hists.py` - Aggregate background distributions of specified variant QC metrics into histograms. The first-pass run determines minimum and maximum values per metric and the second-pass run aggregates QUAL metrics binned by allele frequency and aggregates other metrics over manually set bins/ranges. Writes results out to json files (primarily for browser loading purposes).

### Resources

> **load_data/resources/**

Author: I actually don't think I ever used `dbsnp.header.fixed`, but I can try to look into what it was, I'm assuming it's something Laurent used

* `vqsr.alleleSpecific.header.fixed` - Header supplied to `load_vqsr.py --header_path` to fix incorrect Number on AS_SOR description.

**resources/** - Resources for all gnomAD input and output files at each step of the QC and release pipeline.

## Others
**load_data/**

> * `compute_coverage.py` — Compute coverage statistics for every base of the GRCh38 reference excluding centromeres and telomeres. The following coverage stats are calculated: mean, median, total DP, and fraction of samples with coverage above X, for each x in [1, 5, 10, 15, 20, 25, 30, 50, 100].

Reviewer: ```suggestion
* `compute_coverage.py` — Compute coverage statistics for every base of the GRCh38 reference excluding centromeres and telomeres. The following coverage stats are calculated: mean, median, total DP, and fraction of samples with coverage above x, for each x in [1, 5, 10, 15, 20, 25, 30, 50, 100].
```

* `evaluation.py`
  * `--create_bin_ht` - Create Table with bin annotations based on the ranking of random forest and/or VQSR variant quality scores, with SNPs and indels handled separately. Additional bin annotations are added for the following stratifications: bi-allelic variants, singletons, and bi-allelic singletons.
  * `--create_aggregated_bin_ht` - Compute aggregated metrics for each bin that are useful for comparison of variant filtering performance across multiple random forest models and VQSR.
  * `--extract_truth_samples` - Extract truth samples (NA12878 and synthetic diploid sample) from the full callset MatrixTable for comparison to their truth data.

>   * `--merge_with_truth_data` - Computes a table for each truth sample (NA12878 and synthetic diploid sample) comparing the truth sample in the callset to the truth data.

Reviewer: ```suggestion
  * `--merge_with_truth_data` - Compute a table for each truth sample (NA12878 and synthetic diploid sample) comparing the truth sample in the callset to the truth data.
```

**create_release/**
* `create_release_sites_ht.py` - Combine frequency, filtering allele frequency, variant QC, VEP, dbSNP, in silico scores, and variant QC metric histogram annotations into unified table. Add frequency annotations for each sample subset (e.g., controls/biobanks, non-neuro, non-cancer, non-TOPMed, non-v2, 1KG, HGDP).
* `create_hgdp_tgp_subset.py` - Subset raw sparse MatrixTable to only samples in the Human Genome Diversity Project (HGDP) and 1000 Genomes Project (1KG/TGP) to create an unsplit sparse MatrixTable and split dense MatrixTable (with full sample and variant annotations) that can be made publicly available.
* `make_var_annot_hists.py` - Aggregate background distributions of specified variant QC metrics into histograms. The first-pass run determines minimum and maximum values per metric and the second-pass run aggregates QUAL metrics binned by allele frequency and aggregates other metrics over manually set bins/ranges. Writes results out to json files (primarily for browser loading purposes).

> * `prepare_vcf_data_release.py` - Select and reformat release annotations for VCF export, create VCF header text, and perform sanity check on release annotations.

Reviewer: ```suggestion
* `prepare_vcf_data_release.py` - Select and reformat release annotations for VCF export, create VCF header text, and perform validity checks on release annotations.
```
